Armadillo/src/main/java/com/scribd/armadillo/playback/PlaybackEngine.kt

```diff
@@ -229,4 +246,25 @@ internal class ExoplayerPlaybackEngine(private var audioPlayable: AudioPlayable)
         exoPlayer.seekTo(exoPlayer.currentWindowIndex, position.longValue)
         stateModifier.dispatch(PlaybackProgressAction(position, exoPlayer.playerDuration()))
     }
+
+    private fun clearExistingExoPlayers() {
+        exoplayerReferences.forEach {
+            it.exoPlayer.release()
+            it.exoPlayer.removeListener(it.listener)
+        }
+        exoplayerReferences.clear()
+    }
+
+    private fun initContinuousProgressUpdates(exoPlayer: ExoPlayer) {
+        val exceptionHandler = CoroutineExceptionHandler { _, throwable -> Log.e(TAG, "Continuous Progress Error: ${throwable.message}") }
+        CoroutineScope(Dispatchers.Main + exceptionHandler).launch {
+            while (exoPlayer.playbackState == ExoPlayer.STATE_READY || exoPlayer.playbackState == ExoPlayer.STATE_BUFFERING) {
+                if (exoPlayer.hasProgressAvailable()) {
+                    stateModifier.dispatch(PlaybackProgressAction(exoPlayer.currentPositionInDuration(), exoPlayer.playerDuration()))
```

Comment:
Does this fix the problem of the service that responds to audio controls no longer working, or just the progress stopping?

Reply:
No, just the progress. But the controls may be working in most cases of this error being reported